### ISSUE TITLE
There are some wrong report

### ISSUE THREAD

**Issue by Unlocku** [crytic/slither#308]
The source code of smart contract is :

~~~~
/**
 *Submitted for verification at Etherscan.io on 2019-04-30
*/

pragma solidity ^0.4.25;

/**
 * @title ERC20 interface
 * @dev see https://github.com/ethereum/EIPs/issues/20
 */
interface IERC20 {
  function totalSupply() external view returns (uint256);

  function balanceOf(address who) external view returns (uint256);

  function allowance(address owner, address spender)
    external view returns (uint256);

  function transfer(address to, uint256 value) external returns (bool);

  function approve(address spender, uint256 value)
    external returns (bool);

  function transferFrom(address from, address to, uint256 value)
    external returns (bool);

  event Transfer(
    address indexed from,
    address indexed to,
    uint256 value
  );

  event Approval(
    address indexed owner,
    address indexed spender,
    uint256 value
  );
}



/**
 * @title SafeMath
 * @dev Math operations with safety checks that revert on error
 */
library SafeMath {

  /**
  * @dev Multiplies two numbers, reverts on overflow.
  */
  function mul(uint256 a, uint256 b) internal pure returns (uint256) {
    // Gas optimization: this is cheaper than requiring 'a' not being zero, but the
    // benefit is lost if 'b' is also tested.
    // See: https://github.com/OpenZeppelin/openzeppelin-solidity/pull/522
    if (a == 0) {
      return 0;
    }

    uint256 c = a * b;
    require(c / a == b);

    return c;
  }

  /**
  * @dev Integer division of two numbers truncating the quotient, reverts on division by zero.
  */
  function div(uint256 a, uint256 b) internal pure returns (uint256) {
    require(b > 0); // Solidity only automatically asserts when dividing by 0
    uint256 c = a / b;
    // assert(a == b * c + a % b); // There is no case in which this doesn't hold

    return c;
  }

  /**
  * @dev Subtracts two numbers, reverts on overflow (i.e. if subtrahend is greater than minuend).
  */
  function sub(uint256 a, uint256 b) internal pure returns (uint256) {
    require(b <= a);
    uint256 c = a - b;

    return c;
  }

  /**
  * @dev Adds two numbers, reverts on overflow.
  */
  function add(uint256 a, uint256 b) internal pure returns (uint256) {
    uint256 c = a + b;
    require(c >= a);

    return c;
  }

  /**
  * @dev Divides two numbers and returns the remainder (unsigned integer modulo),
  * reverts when dividing by zero.
  */
  function mod(uint256 a, uint256 b) internal pure returns (uint256) {
    require(b != 0);
    return a % b;
  }
}


/**
 * @title Standard ERC20 token
 *
 * @dev Implementation of the basic standard token.
 * https://github.com/ethereum/EIPs/blob/master/EIPS/eip-20.md
 * Originally based on code by FirstBlood: https://github.com/Firstbloodio/token/blob/master/smart_contract/FirstBloodToken.sol
 */
contract ERC20 is IERC20 {
  using SafeMath for uint256;

  mapping (address => uint256) private _balances;

  mapping (address => mapping (address => uint256)) private _allowed;

  uint256 private _totalSupply;

  /**
  * @dev Total number of tokens in existence
  */
  function totalSupply() public view returns (uint256) {
    return _totalSupply;
  }

  /**
  * @dev Gets the balance of the specified address.
  * @param owner The address to query the balance of.
  * @return An uint256 representing the amount owned by the passed address.
  */
  function balanceOf(address owner) public view returns (uint256) {
    return _balances[owner];
  }

  /**
   * @dev Function to check the amount of tokens that an owner allowed to a spender.
   * @param owner address The address which owns the funds.
   * @param spender address The address which will spend the funds.
   * @return A uint256 specifying the amount of tokens still available for the spender.
   */
  function allowance(
    address owner,
    address spender
   )
    public
    view
    returns (uint256)
  {
    return _allowed[owner][spender];
  }

  /**
  * @dev Transfer token for a specified address
  * @param to The address to transfer to.
  * @param value The amount to be transferred.
  */
  function transfer(address to, uint256 value) public returns (bool) {
    require(value <= _balances[msg.sender]);
    require(to != address(0));

    _balances[msg.sender] = _balances[msg.sender].sub(value);
    _balances[to] = _balances[to].add(value);
    emit Transfer(msg.sender, to, value);
    return true;
  }

  /**
   * @dev Approve the passed address to spend the specified amount of tokens on behalf of msg.sender.
   * Beware that changing an allowance with this method brings the risk that someone may use both the old
   * and the new allowance by unfortunate transaction ordering. One possible solution to mitigate this
   * race condition is to first reduce the spender's allowance to 0 and set the desired value afterwards:
   * https://github.com/ethereum/EIPs/issues/20#issuecomment-263524729
   * @param spender The address which will spend the funds.
   * @param value The amount of tokens to be spent.
   */
  function approve(address spender, uint256 value) public returns (bool) {
    require(spender != address(0));

    _allowed[msg.sender][spender] = value;
    emit Approval(msg.sender, spender, value);
    return true;
  }

  /**
   * @dev Transfer tokens from one address to another
   * @param from address The address which you want to send tokens from
   * @param to address The address which you want to transfer to
   * @param value uint256 the amount of tokens to be transferred
   */
  function transferFrom(
    address from,
    address to,
    uint256 value
  )
    public
    returns (bool)
  {
    require(value <= _balances[from]);
    require(value <= _allowed[from][msg.sender]);
    require(to != address(0));

    _balances[from] = _balances[from].sub(value);
    _balances[to] = _balances[to].add(value);
    _allowed[from][msg.sender] = _allowed[from][msg.sender].sub(value);
    emit Transfer(from, to, value);
    return true;
  }

  /**
   * @dev Increase the amount of tokens that an owner allowed to a spender.
   * approve should be called when allowed_[_spender] == 0. To increment
   * allowed value is better to use this function to avoid 2 calls (and wait until
   * the first transaction is mined)
   * From MonolithDAO Token.sol
   * @param spender The address which will spend the funds.
   * @param addedValue The amount of tokens to increase the allowance by.
   */
  function increaseAllowance(
    address spender,
    uint256 addedValue
  )
    public
    returns (bool)
  {
    require(spender != address(0));

    _allowed[msg.sender][spender] = (
      _allowed[msg.sender][spender].add(addedValue));
    emit Approval(msg.sender, spender, _allowed[msg.sender][spender]);
    return true;
  }

  /**
   * @dev Decrease the amount of tokens that an owner allowed to a spender.
   * approve should be called when allowed_[_spender] == 0. To decrement
   * allowed value is better to use this function to avoid 2 calls (and wait until
   * the first transaction is mined)
   * From MonolithDAO Token.sol
   * @param spender The address which will spend the funds.
   * @param subtractedValue The amount of tokens to decrease the allowance by.
   */
  function decreaseAllowance(
    address spender,
    uint256 subtractedValue
  )
    public
    returns (bool)
  {
    require(spender != address(0));

    _allowed[msg.sender][spender] = (
      _allowed[msg.sender][spender].sub(subtractedValue));
    emit Approval(msg.sender, spender, _allowed[msg.sender][spender]);
    return true;
  }

  /**
   * @dev Internal function that mints an amount of the token and assigns it to
   * an account. This encapsulates the modification of balances such that the
   * proper events are emitted.
   * @param account The account that will receive the created tokens.
   * @param amount The amount that will be created.
   */
  function _mint(address account, uint256 amount) internal {
    require(account != 0);
    _totalSupply = _totalSupply.add(amount);
    _balances[account] = _balances[account].add(amount);
    emit Transfer(address(0), account, amount);
  }

  /**
   * @dev Internal function that burns an amount of the token of a given
   * account.
   * @param account The account whose tokens will be burnt.
   * @param amount The amount that will be burnt.
   */
  function _burn(address account, uint256 amount) internal {
    require(account != 0);
    require(amount <= _balances[account]);

    _totalSupply = _totalSupply.sub(amount);
    _balances[account] = _balances[account].sub(amount);
    emit Transfer(account, address(0), amount);
  }

  /**
   * @dev Internal function that burns an amount of the token of a given
   * account, deducting from the sender's allowance for said account. Uses the
   * internal burn function.
   * @param account The account whose tokens will be burnt.
   * @param amount The amount that will be burnt.
   */
  function _burnFrom(address account, uint256 amount) internal {
    require(amount <= _allowed[account][msg.sender]);

    // Should https://github.com/OpenZeppelin/zeppelin-solidity/issues/707 be accepted,
    // this function needs to emit an event with the updated approval.
    _allowed[account][msg.sender] = _allowed[account][msg.sender].sub(
      amount);
    _burn(account, amount);
  }
}


/**
 * @title Vtoken
 * @dev Very simple ERC20 Token example, where all tokens are pre-assigned to the creator.
 * Note they can later distribute these tokens as they wish using `transfer` and other
 * `ERC20` functions.
 */
contract Vtoken is ERC20 {

  string public constant name = "Vtoken";
  string public constant symbol = "vtoken";
  uint8 public constant decimals = 18;

  uint256 public constant INITIAL_SUPPLY = 100000000000 * (10 ** uint256(decimals));

  /**
   * @dev Constructor that gives msg.sender all of existing tokens.
   */
  constructor() public {
    _mint(msg.sender, INITIAL_SUPPLY);
  }

}
~~~~

and I just use slither to scan this contract, the report is
~~~~
INFO:Detectors:

IERC20.totalSupply (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#12) should be declared external
ERC20.totalSupply (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#126-128) should be declared external
ERC20.balanceOf (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#135-137) should be declared external
IERC20.balanceOf (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#14) should be declared external
IERC20.allowance (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#16-17) should be declared external
ERC20.allowance (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#145-154) should be declared external
ERC20.transfer (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#161-169) should be declared external
IERC20.transfer (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#19) should be declared external
IERC20.approve (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#21-22) should be declared external
ERC20.approve (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#180-186) should be declared external
ERC20.transferFrom (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#194-211) should be declared external
IERC20.transferFrom (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#24-25) should be declared external
ERC20.increaseAllowance (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#222-235) should be declared external
ERC20.decreaseAllowance (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#246-259) should be declared external
Reference: https://github.com/crytic/slither/wiki/Detector-Documentation#public-function-that-could-be-declared-as-external
INFO:Detectors:
Function 'ERC20._mint' (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#268-273) is not in mixedCase
Function 'ERC20._burn' (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#281-288) is not in mixedCase
Function 'ERC20._burnFrom' (..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#297-305) is not in mixedCase
Reference: https://github.com/crytic/slither/wiki/Detector-Documentation#conformance-to-solidity-naming-conventions
INFO:Slither:..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol analyzed (4 contracts), 17 result(s) found
~~~~

Such as the first report
~~~~
(..\..\contract\contract_1\0x9d8e0b763e43975008528ea2e18ba547c40722e1.sol#12) should be declared external
ERC20.totalSupply 
~~~~

I think it means the function as blow should be external, but It is external, so why does is report? 
~~~~
function totalSupply() external view returns (uint256);
~~~~

**Comment by montyly** [comment]
Hi @Unlocku 

Thank you for reporting the issue. It is indeed a bug that happens when a function is first defined as external in a base contract, and then re-define as public in a contract inheriting the base contract.

Smaller poc:
```
contract C{

    function f() external;

}

contract D is C{

    function f() public{
    }

}
```

We will fix it.

**Comment by montyly** [comment]
This seems to be fixed in `0.6.7`. Only the correct `contract.function` are reported now.
